### PR TITLE
Bugfix DPL raw proxy: fixed an indexing problem

### PR DESCRIPTION
**Pull request by matthiasrichter**
Fixes O2-2742

Enables the support for sequence of payloads/pruned headers originally
committed in 74dc3e704132fb0da641c34777337c87e5840b8a

There was a remnant check which was not adapted to the new meaning of
message index, causing sporadic access beyond the range of the incoming
FairMQPart object.

This reverts commits
   aa64ad5df402b6af2566748b6a03c1bc092231e3.
   d6d40712986b149a0dcd253ed31e5edb735eeab8.
   078f632ac6086e6e7400d2b10c60049a1e172cb3.

### CODE REVIEW

File: Framework/Core/src/ExternalFairMQDeviceProxy.cxx [hunk 1]
```diff
@@ -221,69 +223,81 @@ InjectorFunction dplModelAdaptor(std::vector<OutputSpec> const& filterSpecs, boo
...
     std::unordered_map<std::string, FairMQParts> outputs;
     std::vector<std::string> unmatchedDescriptions;
-    int lastSplitPartIndex = -1;
-    std::string channelNameForSplitParts;
     static int64_t dplCounter = -1;
     dplCounter++;
-    for (size_t msgidx = 0; msgidx < parts.Size() / 2; ++msgidx) {
-      const auto dh = o2::header::get<DataHeader*>(parts.At(msgidx * 2)->GetData());
+    for (size_t msgidx = 0; msgidx < parts.Size(); msgidx += 2) {
+      const auto dh = o2::header::get<DataHeader*>(parts.At(msgidx)->GetData());
```
Comment (matthiasrichter) [inline Framework/Core/src/ExternalFairMQDeviceProxy.cxx:229]: This line, and the same below, was the problematic line which was not adapted in 74dc3e704132fb0da641c34777337c87e5840b8a

**Comment by ktf** [comment]
This is fine with me, but again I am waiting @davidrohr opinion on wether it's safe to merge this now.

**Comment by matthiasrichter** [comment]
we clearly need @davidrohr to approve this

**Comment by matthiasrichter** [comment]
and,  @TimoWilken, can the fullCI test be triggered again

**Comment by TimoWilken** [comment]
> and, @TimoWilken, can the fullCI test be triggered again

Done!

**Comment by davidrohr** [comment]
Great, thx, will test it later today (there was a mountain of requests what I shall test today, so spending the day compiling... :))

**Review by davidrohr — APPROVED** [review]
Ran this through Pb-Pb full system test starting from DataDistribution and looks good now

**Pull request merged**